test/test_fast_float_distance.cpp

```diff
@@ -0,0 +1,47 @@
+//  (C) Copyright John Maddock 2008.
+//  (C) Copyright Matt Borland 2022.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <boost/math/concepts/real_concept.hpp>
+#include <boost/test/tools/floating_point_comparison.hpp>
+#include <boost/math/special_functions/next.hpp>
+#include <boost/math/special_functions/ulp.hpp>
+#include <boost/math/special_functions/fast_float_distance.hpp>
+#include <boost/math/tools/assert.hpp>
+#include <boost/multiprecision/float128.hpp>
+
+#include "math_unit_test.hpp"
+
+
+template <class T>
+void test_value(const T& val)
+{
+    using namespace boost::math;
+
+    BOOST_MATH_ASSERT(fast_float_distance(float_next(val), val) == -1);
+    BOOST_MATH_ASSERT(float_next(val) > val);
+    BOOST_MATH_ASSERT(float_next(float_prior(val)) == val);
+
+    BOOST_MATH_ASSERT(fast_float_distance(float_advance(val, 4), val) == -4);
+    BOOST_MATH_ASSERT(fast_float_distance(float_advance(val, -4), val) == 4);
+    if(std::numeric_limits<T>::is_specialized && (std::numeric_limits<T>::has_denorm == std::denorm_present))
```

Comment:
Doesn't the fast float distance unconditionally require denorm support?

Reply:
It's just type punning so as long as the floating point and fixed width integer are the same number of bits you should be fine. I don't know of a platform without denorm support to test on.